pkg/karmadactl/taint_test.go

```diff
@@ -0,0 +1,585 @@
+package karmadactl
+
+import (
+	"reflect"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+
+	clusterv1alpha1 "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
+)
+
+func TestValidate(t *testing.T) {
+	cases := []struct {
+		taintOpt    CommandTaintOption
+		description string
+		expectFatal bool
+	}{
+		{
+			taintOpt:    CommandTaintOption{resources: []string{"cluster"}},
+			description: "Cluster name is not provided",
+			expectFatal: true,
+		},
+		{
+			taintOpt:    CommandTaintOption{resources: []string{"cluster", "cluster-name"}},
+			description: "Cluster name is provided",
+			expectFatal: false,
+		},
+	}
+	for _, c := range cases {
+		sawFatal := false
+		err := c.taintOpt.Validate()
+		if err != nil {
+			sawFatal = true
+		}
+		if c.expectFatal {
+			if !sawFatal {
+				t.Fatalf("%s expected not to fail", c.description)
+			}
+		}
+	}
+}
+
+func TestParseTaintArgs(t *testing.T) {
+	cases := []struct {
+		taintOpt    CommandTaintOption
+		taintArgs   []string
+		description string
+		expectFatal bool
+	}{
+		{
+			taintOpt:    CommandTaintOption{},
+			taintArgs:   []string{"cluster", "cluster_name", "foo=bar:NoSchedule", "foo_1=bar:NoExecute"},
+			description: "Correct order of arguments",
+			expectFatal: false,
+		},
+		{
+			taintOpt:    CommandTaintOption{},
+			taintArgs:   []string{"cluster", "foo=bar:NoSchedule", "cluster_name", "foo_1=bar:NoExecute"},
+			description: "Incorrect order of arguments",
+			expectFatal: true,
+		},
+	}
+
+	for _, c := range cases {
+		sawFatal := false
+		_, err := c.taintOpt.parseTaintArgs(c.taintArgs)
```

Comment:
when err is nil, we can judge with the return arrays.

Reply:
This function only classify args by order, the return arrays could not to be checked. If check arrays, we need to call below way again.

```
for _, taint := range taints {
  if strings.Contains(taint, "=") || strings.Contains(taint, ":") || strings.HasSuffix(taint, "-") {
	  continue
  }
}
```
It could be loop test.